yt_dlp/extractor/tencent.py

```diff
@@ -100,8 +119,8 @@ def _extract_video_native_subtitles(self, api_response, subtitles_format):
     def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
         formats, subtitles = [], {}
         for video_format, subtitle_format, video_quality in (
-                # '': 480p, 'shd': 720p, 'fhd': 1080p
-                ('mp4', 'srt', ''), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd')):
+                # 'sd': 270p, 'hd': 480p, 'shd': 720p, 'fhd': 1080p, 'uhd': 4k, 'hdr10': 'HDR10'
+                ('mp4', 'srt', 'sd'), ('mp4', 'srt', 'hd'), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd'), ('hls', 'vtt', 'uhd'), ('hls', 'vtt', 'hdr10')):
```

Comment:
I remember only extracting 480p, 720p and 1080p  because doing so many requests will get you blocked.

If you really want all formats, maybe we can add an extractor argument for this? Not all videos have 4k and I think it's not worth it getting blocked.

Reply:
Most new videos on v.qq.com now have 4K (since v.qq.com likes to upsample videos, some older videos may also have 4K)
In response to the blocking issue, I could remove 270P, but having 4K by default would make players like mpv easy.

Reply:
Agree, 270p doesn't make much sense

I see your point, but you have to consider that this code also needs to work for WeTv and Iflix non-authenticated users. As far as I know, you can pass args to mpv (--ytdl-raw-options, https://mpv.io/manual/master/)

Reply:
I updated `_extract_all_video_formats_and_subtitles` implementation, it will now only fetch the quality it can.

Reply:
Well, there is something wrong with the new implementation, and I'll keep trying to fix it.

Reply:
I fixed some bugs and now it will only fetch the quality it can.